src/fitnesse/slim/instructions/SystemExitSecurityManager.java

```diff
@@ -26,9 +26,19 @@ private SystemExitSecurityManager(SecurityManager delegate) {
   public static void activateIfWanted() {
     if (isPreventSystemExit()) {
       SecurityManager currentSecMgr = System.getSecurityManager();
-      SystemExitSecurityManager systemExitSecurityManager = new SystemExitSecurityManager(
+      if (currentSecMgr != null) {
```

Comment:
I believe the security manager will be null in plain java also, therefore the system exit one will not be added.
What I've seen (in slf4j) is a check for android by getting system property: `java.vendor.url` turning that into lowercase and checking whether it contains `android`...